libsolidity/formal/SMTEncoder.cpp

```diff
@@ -787,7 +787,6 @@ void SMTEncoder::visitAddMulMod(FunctionCall const& _funCall)
 	auto x = expr(*args.at(0));
 	auto y = expr(*args.at(1));
 	auto k = expr(*args.at(2));
-	m_context.addAssertion(k != 0);
```

Comment:
I think these assertions should also be removed from `BMC`

Reply:
True, I will check that.
Also, I have been wondering about the UnderOverflow verification target. I have eliminated its use from CHC engine, maybe I should try to do the same in BMC engine and remove this type of VerificationTarget?

Reply:
I think for BMC it's actually easier to keep that target because the ways the verification targets are built/used

Reply:
It looks like these assertion about the second argument of division not being zero were only in the SMTEncoder, not in the subclasses, no? So there are no more places where they can be removed from.

Reply:
Ah yes, true!